### PR TITLE
SPL Filter now working correctly 

### PR DESCRIPTION
**Pull request by jdtoombs**
Mikaela made a comment that the SPL filter was returning SPP results. Renamed the SPP filter accordingly and it now only return SPL properties. Backend now reflects frontend!

Also my [old PR that has been approved](https://github.com/bcgov/PIMS/pull/956) has been updated and I have re-requested some reviews as there has been some changes.

splfix

### CODE REVIEW

File: backend/dal/Helpers/Extensions/PropertyExtensions.cs [hunk 1]
```diff
@@ -95,8 +95,6 @@ private static IQueryable<Entity.Views.Property> GenerateCommonQuery(this PimsCo
                 query = query.Where(p => p.ProjectNumbers.Contains(filter.ProjectNumber));
             if (filter.IgnorePropertiesInProjects == true)
                 query = query.Where(p => p.ProjectNumbers == null || p.ProjectNumbers == "[]");
-            if (filter.InSurplusPropertyProgram == true)
```
Comment (jdtoombs) [inline backend/dal/Helpers/Extensions/PropertyExtensions.cs:98]: I think at one point this was the desired filter but it evolved over time and only got changed on the FE

**Review by goemen — APPROVED** [review]
Looks good

File: frontend/src/components/common/form/SppButton.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ const SppButton: React.FC<ISppButtonProps> = ({
 
   const iconColor = inEnhancedReferralProcess
```
Comment (agalbrai) [inline frontend/src/components/common/form/SppButton.tsx:30]: I think this carried over from the old button... The colour doesn't actually change anymore

Comment (agalbrai) [reply frontend/src/components/common/form/SppButton.tsx:30]: I just checked, also the tooltip doesn't work anymore

**Review by agalbrai — APPROVED** [review]
Thank goodness for "Find and Replace" haha

File: frontend/src/components/common/form/SppButton.tsx (hunk 2 above)
Comment (jdtoombs) [reply frontend/src/components/common/form/SppButton.tsx:30]: Oh you're right, I think I vaguely remember the old button lol, looks like it is not used anywhere but the test now! Will remove

File: backend/dal/Helpers/Extensions/PropertyExtensions.cs [hunk 3]
```diff
@@ -159,6 +157,18 @@ private static IQueryable<Entity.Views.Property> GenerateCommonQuery(this PimsCo
                             && property.ProjectNumbers.Contains(project.ProjectNumber)));
             }
 
+            if (filter.InSurplusPropertyList.HasValue && filter.InSurplusPropertyList.Value)
```
Comment (Fosol) [inline backend/dal/Helpers/Extensions/PropertyExtensions.cs:160]: I think you can just go 
```
filter.InSuplusPropertyList == true
```

File: backend/dal/Helpers/Extensions/PropertyExtensions.cs [hunk 4]
```diff
@@ -159,6 +157,18 @@ private static IQueryable<Entity.Views.Property> GenerateCommonQuery(this PimsCo
...
+            {
+                var statuses = context.Workflows.Where(w => w.Code == "SPL")
+                    .SelectMany(w => w.Status).Where(x => !x.Status.IsTerminal)
+                    .Select(x => x.StatusId).Distinct().ToArray();
+
+                query = query.Where(property =>
```
Comment (Fosol) [inline backend/dal/Helpers/Extensions/PropertyExtensions.cs:166]: This is a very expensive query.   If when we search for ERP or SPL I think we should invert the original query.  Start with projects and return properties form that list.  Then we can apply the other filter values to the properties.

Comment (jdtoombs) [reply backend/dal/Helpers/Extensions/PropertyExtensions.cs:166]: Ahh ok, I'll switch the ERP one as well

**Pull request closed without merging**